quinn/src/connection.rs

```diff
@@ -382,7 +386,18 @@ impl Connection {
         }
     }
 
-    /// Opens a (Multi)Path
+    /// Opens an additional path if the multipath extension is negotiated.
+    ///
+    /// The returned future completes once the path is either fully opened and ready to
+    /// carry application data, or if there was an error.
+    ///
+    /// Dropping the returned future does not cancel the opening of the path, the
+    /// [`PathEvent::Opened`] event will still be emitted from [`Self::path_events`] if the
+    /// path opens.  The [`PathId`] for the events can be extracted from
+    /// [`OpenPath::path_id`].
+    ///
+    /// If opening fails either the future will be immediately ready, or another
+    /// [`PathEvent`] for the [`PathId`] will be emitted, indicating the failure.
```

Comment:
This can fail in a non-immediate way too. Are you simply trying to say thta failures are also emitted as events?